src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -65,15 +69,14 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
             }
         }
 
-        private static ConstructorInitializerSyntax GenerateBaseConstructorInitializer(ImmutableArray<IMethodSymbol> baseTypeConstructors)
+        private static ConstructorInitializerSyntax GenerateBaseConstructorInitializer(IMethodSymbol baseTypeConstructor)
         {
             ConstructorInitializerSyntax constructorInitializer = SyntaxFactory.ConstructorInitializer(SyntaxKind.BaseConstructorInitializer);
 
-            IMethodSymbol baseConstructor = baseTypeConstructors.OrderBy(c => c.Parameters.Count()).First();
-            foreach (IParameterSymbol parameter in baseConstructor.Parameters)
+            foreach (IParameterSymbol parameter in baseTypeConstructor.Parameters)
             {
                 IdentifierNameSyntax identifier;
-                if (parameter.Type.IsValueType)
+                if (parameter.Type.IsValueType || parameter.NullableAnnotation == NullableAnnotation.Annotated)
```

Comment:
Please add a code comment to explain what this conditions checks as it isn't obvious.

Reply:
added a comment